go-controller/pkg/node/gateway_iptables.go

```diff
@@ -62,9 +62,10 @@ type iptRule struct {
 	protocol iptables.Protocol
 }
 
-func addIptRules(rules []iptRule) error {
+func addIptRules(rules []iptRule, append bool) error {
```

Comment:
nit: can we split this into two functions and eliminate the boolean?
maybe `appendIptRules(...)` and `insertIptRulesAsFirst()`

Reply:
So I am going with what @trozet  said here: code readability here and trying to minimize the code duplication. I am creating `appendIptRules` and `insertIptRules` which call addIptRules with either the bool=true so append or bool = false prepend. At the code call from other parts of code have more readable and easy to digest names then!